api/v1/storagecluster_types.go

```diff
@@ -247,14 +247,32 @@ type ManageCephBlockPools struct {
 }
 
 // ManageCephNonResilientPools defines how to reconcile ceph non-resilient pools
+// +kubebuilder:validation:XValidation:rule="!(self.useExistingOsds == true) || !has(self.resources)",message="resources should not be set when useExistingOsds is true"
+// +kubebuilder:validation:XValidation:rule="!(self.useExistingOsds == true) || !has(self.count)",message="count should not be set when useExistingOsds is true"
+// +kubebuilder:validation:XValidation:rule="!(self.useExistingOsds == true) || !has(self.volumeClaimTemplate)",message="volumeClaimTemplate should not be set when useExistingOsds is true"
 type ManageCephNonResilientPools struct {
 	Enable bool `json:"enable,omitempty"`
+	// UseExistingOsds is set when pools has to use existing OSDs
+	// +kubebuilder:validation:XValidation:rule="oldSelf == null || self == oldSelf",message="useExistingOsds field cannot be modified once set"
+	UseExistingOsds *bool `json:"useExistingOsds,omitempty"`
+	// PoolNamePrefix is prefix string used in the replica-1 pools names
+	// +kubebuilder:validation:MaxLength=200
+	// +kubebuilder:validation:XValidation:rule="oldSelf == null || self == oldSelf",message="poolNamePrefix field cannot be modified once set"
+	// +optional
+	PoolNamePrefix *string `json:"poolNamePrefix,omitempty"`
+	// SkipCreateStorageclass if set will skip the default replica-1 topological storageclass creation
+	// +kubebuilder:validation:XValidation:rule="oldSelf == null || self == oldSelf",message="skipCreateStorageclass field cannot be modified once set"
```

Comment:
Why can't ths skipCreateStorageClass be modified? Whether to create the storage class seems fine to change.